pkg/system/phase2_creating.go

```diff
@@ -353,3 +418,88 @@ func (r *Reconciler) SetDesiredAgentProfile(profileString string) string {
 	util.Panic(err)
 	return string(profileBytes)
 }
+
+// fixPvs make changes related to pvs if an upgrade process is needed
+func (r *Reconciler) fixPvs() {
+
+	// declare the old pvc
+	oldPvc := &corev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{Kind: "PersistentVolumeClaim"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "db-" + options.SystemName + "-core-0",
+			Namespace: options.Namespace,
+		},
+	}
+	// declare the new pvc
+	newPvc := &corev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{Kind: "PersistentVolumeClaim"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "db-" + options.SystemName + "-db-0",
+			Namespace: options.Namespace,
+		},
+	}
+	oldPvcExist := util.KubeCheck(oldPvc)
+	newPvcExist := util.KubeCheck(newPvc)
+	pvName := ""
+	// find pvName
+	if oldPvcExist {
+		pvName = oldPvc.Spec.VolumeName
+	} else if newPvcExist {
+		pvName = newPvc.Spec.VolumeName
+	}
+	// declare the pv
+	pv := &corev1.PersistentVolume{
+		TypeMeta: metav1.TypeMeta{Kind: "PersistentVolume"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pvName,
+		},
+	}
+	// check if old pvc exist
+	if oldPvcExist {
+		r.Logger.Infof("Old version found , start updating.")
+
+		// delete old noobaa-core statefulset
+		oldSts := &appsv1.StatefulSet{
+			TypeMeta: metav1.TypeMeta{Kind: "StatefulSet"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "noobaa-core",
+				Namespace: options.Namespace,
+			},
+		}
+		util.KubeCheck(oldSts)
+		util.KubeDelete(oldSts)
```

Comment:
Sounds a bit weird that this is done before setting the Retain reclaim policy on the pv, no?